### PR TITLE
Avoid toggling favourite for all selected files

### PR DESCRIPTION
**Pull request by mario**
We don't want to toggle favorite for files we know are already favorited. 
This simple change avoids this for both singular and multiple favorite
changes.

(And yes, there's some redundancy because toggleFavorite will do the required
check even if called  from toggleFavorites - but at least there are no redundant method
calls)

There are other problems in the way toggling a favorite is presented in the UI, but I'll leave
that to one of my future PRs.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -10,7 +10,7 @@ buildscript {
         jcenter()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:2.2.1'
+        classpath 'com.android.tools.build:gradle:2.2.3'
```
Comment (tobiasKaminsky) [inline build.gradle]: This should be in a separate PR.

File: src/com/owncloud/android/ui/helpers/FileOperationsHelper.java [hunk 2]
```diff
@@ -546,26 +547,37 @@ public void syncFile(OCFile file) {
     }
 
     public void toggleFavorites(Collection<OCFile> files, boolean isFavorite){
+        List<OCFile> found = new ArrayList<>();
```
Comment (tobiasKaminsky) [inline src/com/owncloud/android/ui/helpers/FileOperationsHelper.java]: maybe a better name is "alreadyRightState" ore similar?

File: src/com/owncloud/android/ui/helpers/FileOperationsHelper.java [hunk 3]
```diff
@@ -546,26 +547,37 @@ public void syncFile(OCFile file) {
...
+        for(OCFile file : files){
+            if(file.isFavorite() == isFavorite){
+                found.add(file);
+            }
+        }
+
+        files.removeAll(found);
+
```
Comment (tobiasKaminsky) [inline src/com/owncloud/android/ui/helpers/FileOperationsHelper.java]: What a pitty that we cannot use java 8:
  files.removeIf(f -> f.isFavorite() == isFavorite);

File: src/com/owncloud/android/ui/helpers/FileOperationsHelper.java [hunk 4]
```diff
@@ -546,26 +547,37 @@ public void syncFile(OCFile file) {
...
-                mFileActivity,
-                file,
-                mFileActivity.getAccount(),
-                isFavorite);
-        mFileActivity.startService(observedFileIntent);
-
-        /// immediate content synchronization
-        if (file.isFavorite()) {
-            syncFile(file);
+        if (file.isFavorite() != isFavorite) {
```
Comment (tobiasKaminsky) [inline src/com/owncloud/android/ui/helpers/FileOperationsHelper.java]: If you filter all "file.isFavorite() == isFavorite)" in toggleFavorites and toggleFavorite is always called with the opposite favorite state, this comparison should always be true, or?

File: build.gradle (hunk 1 above)
Comment (mario) [inline build.gradle]: Really? Want me to revert this?

File: src/com/owncloud/android/ui/helpers/FileOperationsHelper.java (hunk 2 above)
Comment (mario) [inline src/com/owncloud/android/ui/helpers/FileOperationsHelper.java]: Maybe ... :P

File: src/com/owncloud/android/ui/helpers/FileOperationsHelper.java (hunk 3 above)
Comment (mario) [inline src/com/owncloud/android/ui/helpers/FileOperationsHelper.java]: Heh, yea.

File: src/com/owncloud/android/ui/helpers/FileOperationsHelper.java (hunk 4 above)
Comment (mario) [inline src/com/owncloud/android/ui/helpers/FileOperationsHelper.java]: I do this because there are other places that call togglefavorite directly.

Comment (tobiasKaminsky) [inline src/com/owncloud/android/ui/helpers/FileOperationsHelper.java]: Ok :)

File: src/com/owncloud/android/ui/helpers/FileOperationsHelper.java (hunk 2 above)
Comment (tobiasKaminsky) [inline src/com/owncloud/android/ui/helpers/FileOperationsHelper.java]: It was just an idea as found is, at least for me, too generic.
But is not a big deal if you leave it.

Comment (mario) [inline src/com/owncloud/android/ui/helpers/FileOperationsHelper.java]: Yup, I agree - I'll change this (but can't now, internet is very bad)

File: build.gradle (hunk 1 above)
Comment (tobiasKaminsky) [inline build.gradle]: According to our policy, which we had the last years, one PR should only solve one issue.
And for tracking the changes it is easier if this is in a separate PR.

File: src/com/owncloud/android/ui/helpers/FileOperationsHelper.java (hunk 3 above)
Comment (AndyScherzinger) [inline src/com/owncloud/android/ui/helpers/FileOperationsHelper.java]: Can't we move to Android 8 in another PR? Is imho supported but would probably require JACK activated  and also some work with the Android Drone docker image.

Comment (mario) [inline src/com/owncloud/android/ui/helpers/FileOperationsHelper.java]: You probably mean Java 8 hehe :) We can, it's not too hard to do it but probably not a priority atm :)

**Comment by mario** [comment]
@tobiasKaminsky two relevant comments of yours are now improved :)

File: src/com/owncloud/android/ui/helpers/FileOperationsHelper.java (hunk 3 above)
Comment (AndyScherzinger) [inline src/com/owncloud/android/ui/helpers/FileOperationsHelper.java]: Not at the moment, no. But if we have the time (at some point) we should probably do it :) 
--> https://github.com/nextcloud/android/issues/584

File: src/com/owncloud/android/ui/helpers/FileOperationsHelper.java [hunk 5]
```diff
@@ -546,26 +547,37 @@ public void syncFile(OCFile file) {
     }
 
     public void toggleFavorites(Collection<OCFile> files, boolean isFavorite){
+        List<OCFile> alreadyRightStateList = new ArrayList<>();
+        for(OCFile file : files){
```
Comment (AndyScherzinger) [inline src/com/owncloud/android/ui/helpers/FileOperationsHelper.java]: Just formatting, please add a space before the bracket

File: src/com/owncloud/android/ui/helpers/FileOperationsHelper.java [hunk 6]
```diff
@@ -546,26 +547,37 @@ public void syncFile(OCFile file) {
...
+            if(file.isFavorite() == isFavorite){
```
Comment (AndyScherzinger) [inline src/com/owncloud/android/ui/helpers/FileOperationsHelper.java]: Same here just formatting, please add a space before the bracket

**Comment by AndyScherzinger** [comment]
Just a minor formatting comment :) Other than that looks good :)

File: src/com/owncloud/android/ui/helpers/FileOperationsHelper.java (hunk 6 above)
Comment (mario) [inline src/com/owncloud/android/ui/helpers/FileOperationsHelper.java]: This is the same thing as above? :D

Comment (mario) [inline src/com/owncloud/android/ui/helpers/FileOperationsHelper.java]: Meh yea, it wasn't ... ignore me, it's early.

**Comment by AndyScherzinger** [comment]
👍 I'll see that I get it test driven this evening.

**Comment by tobiasKaminsky** [comment]
👍

**Comment by tobiasKaminsky** [comment]
@AndyScherzinger can you give us here (again) a 👍 and then merge?

**Comment by AndyScherzinger** [comment]
👍 should I merge? Just asking since @mario assigned it to 1.5.0. I don't mind putting it into 1.4.2

**Comment by mario** [comment]
@AndyScherzinger it's fine to make this 1.4.2
